File: app/src/main/java/me/lukeforit/spaceofaday/ui/base/DIActivity.java
package me.lukeforit.spaceofaday.ui.base;

import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.google.android.gms.analytics.HitBuilders;
import com.google.android.gms.analytics.Tracker;

import javax.inject.Inject;

import dagger.android.AndroidInjection;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;
import me.lukeforit.spaceofaday.common.SpaceApp;

public abstract class DIActivity extends AppCompatActivity implements HasSupportFragmentInjector {

    private Tracker tracker;

    @Inject
    DispatchingAndroidInjector<Fragment> fragmentDispatchingAndroidInjector;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);

        SpaceApp application = (SpaceApp) getApplication();
        tracker = application.getDefaultTracker();
    }

    @Override
    protected void onResume() {
        super.onResume();
        tracker.setScreenName("Activity-" + this.getClass().getSimpleName());
        tracker.send(new HitBuilders.ScreenViewBuilder().build());
    }

    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return fragmentDispatchingAndroidInjector;
    }
}

File: app/src/main/java/me/lukeforit/spaceofaday/ui/base/DIActivity.kt
package me.lukeforit.spaceofaday.ui.base

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.appcompat.app.AppCompatActivity

import com.google.android.gms.analytics.HitBuilders
import com.google.android.gms.analytics.Tracker

import javax.inject.Inject

import dagger.android.AndroidInjection
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.HasSupportFragmentInjector
import me.lukeforit.spaceofaday.common.SpaceApp

abstract class DIActivity : AppCompatActivity(), HasSupportFragmentInjector {

    private var tracker: Tracker? = null

    @Inject
    internal lateinit var fragmentDispatchingAndroidInjector: DispatchingAndroidInjector<androidx.fragment.app.Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)

        val application = application as SpaceApp
        tracker = application.defaultTracker
    }

    override fun onResume() {
        super.onResume()
        tracker!!.setScreenName("Activity-" + this.javaClass.simpleName)
        tracker!!.send(HitBuilders.ScreenViewBuilder().build())
    }

    override fun supportFragmentInjector(): AndroidInjector<androidx.fragment.app.Fragment>? {
        return fragmentDispatchingAndroidInjector
    }
}
